### PR TITLE
Custom api and worker

### PR DESCRIPTION
**Pull request by madhavajay**
## Description
Please include a summary of the change, the motivation, and any additional context that will help others understand your PR. If it closes one or more open issues, [please tag them as described here](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

## Affected Dependencies
List any dependencies that are required for this change.

## How has this been tested?
- Describe the tests that you ran to verify your changes.
- Provide instructions so we can reproduce.
- List any relevant details for your test configuration.

## Checklist
- [ ] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [ ] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [ ] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [ ] My changes are covered by tests

### CODE REVIEW

File: packages/syft/src/syft/service/api/api_service.py [hunk 1]
```diff
@@ -221,6 +229,145 @@ def api_endpoints(
...
+        # relative
+        from ..job.job_stash import JobStatus
+
+        # So result is a Job object
+        job = result
+        job_service = context.node.get_service("jobservice")
+        job_id = job.id
+        # Question: For a small moment, when job status is updated, it doesn't return the job during the .get() as if
+        # it's not in the stash. Then afterwards if appears again. Is this a bug?
+        timeout = 30  # 10 is too slow on GitHub CI macos?
```
Comment (madhavajay) [inline packages/syft/src/syft/service/api/api_service.py]: How can we handle this? A fixed timeout on a job of unknown length probably won't work, but no timeout might also be a problem.

**Comment by madhavajay** [comment]
Solid work everyone!
Another great party PR  🎉

File: packages/syft/src/syft/service/api/api_service.py (hunk 1 above)
Comment (shubham3121) [reply packages/syft/src/syft/service/api/api_service.py]: Should we pass timeout as an argument to the api with a considerable default timeout of let's say 300sec ?

Comment (IonesioJunior) [reply packages/syft/src/syft/service/api/api_service.py]: [0c5a372](https://github.com/OpenMined/PySyft/pull/8693/commits/0c5a37220b600effed51e82de8ccb5390d915045)
I just added a new field to the TwinAPIEndpoint settings so you can set different timeout values for each endpoint. Now, it's up to the admin to tweak these values whenever needed to keep things running smoothly.

File: packages/syft/src/syft/service/api/api_service.py [hunk 2]
```diff
@@ -316,9 +316,9 @@ def _call_in_jobs(
...
             job = job_service.get(context, job_id)
             time.sleep(0.1)
-            if (time.time() - timeout) > start:
+            if (time.time() - custom_endpoint.endpoint_timeout) > start:
                 return SyftError(
-                    message=f"Function timed out in {timeout} seconds. Get the Job with id: {job_id} to check results."
+                    message=f"Function timed out in {custom_endpoint.endpoint_timeout} seconds. Get the Job with id: {job_id} to check results."
                 )
 
         if job.status == JobStatus.COMPLETED:
```
Comment (madhavajay) [inline packages/syft/src/syft/service/api/api_service.py:324]: Add error to job.result and return the job?

File: packages/syft/src/syft/service/api/api_service.py (hunk 1 above)
Comment (madhavajay) [reply packages/syft/src/syft/service/api/api_service.py]: Excellent! @IonesioJunior ready for merge?

**Review by IonesioJunior — APPROVED** [review]

**Pull request merged**